pallets/system/src/header_builder.rs

```diff
@@ -62,21 +62,22 @@ pub trait HeaderExtensionBuilder {
 		app_extrinsics: Vec<AppExtrinsic>,
 		data_root: H256,
 		block_length: BlockLength,
+		block_number: u32,
 	) -> HeaderExtension;
 
 	/// Generates a random seed using the _epoch seed_ and the _current block_ returned by
 	/// `T::Randomness` type.
 	fn random_seed<T: Config>() -> Seed {
-		let (epoch_seed, block_number) = <T as Config>::Randomness::random_seed();
-		let seed = <T as Config>::Hashing::hash_of(&(&epoch_seed, &block_number));
-
-		log::trace!(
-			target: LOG_TARGET,
-			"Header builder seed {:?} from epoch seed {:?} and block {:?}",
-			seed,
-			epoch_seed,
-			block_number
-		);
+		let seed = if cfg!(feature = "secure_padding_fill") {
```

Comment:
See https://github.com/availproject/engineering/issues/52 to get more info about `secure_padding_fill`

Reply:
Thanks !